packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -49,32 +49,54 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
         return name.replace("@", "").replace("/", "-");
     }
 
+    /**
+     * Take an unknown error object and add the appropriate info from it to the event
+     * NOTE - message and stack will be copied over from the error object,
+     * along with other telemetry properties if it's an ILoggingError
+     * @param event - Event being logged
+     * @param error - Error to extract info from
+     * @param fetchStack - Whether to fetch the current callstack if error.stack is undefined
+     */
     public static prepareErrorObject(event: ITelemetryBaseEvent, error: any, fetchStack: boolean) {
-        if (error === null || typeof error !== "object") {
+        if (typeof error !== "object" || error === null) {
             event.error = error;
         } else {
-            // WARNING: Exceptions can contain PII!
-            // For example, XHR will throw object derived from Error that contains config information
-            // for failed request, including all the headers, and thus - user tokens!
-            // Extract only call stack, message, and couple network-related properties form error object
-
-            const errorAsObject = error as {
-                stack?: string;
-                message?: string;
-            };
-
+            const errorAsObject = error as Partial<Error>;
             event.stack = errorAsObject.stack;
```

Comment:
I think you do not need message & stack because getTelemetryProperties() returns them and I think it's better to have single place to handle it as it might not be possible to log it (see my comment in there)

Reply:
Whoops we forgot that this is copying `message` -> `error` anyway.  That line needs to stay.

Reply:
I do not follow. stack property is enumerated below as getTelemetryProperties() will include it. What's the point to do so here, and get overwritten later? Note that next developer might make mistake of removing it from getTelemetryProperties() (if such need arises) and not realize that it's mostly noop due to code here.

Reply:
For this PR, I'm leaving it as-is, to preserve (incorrect) behavior where `event.stack` and `event.error` are direct copies of `error.stack` and `error.message`. I completely agree with all your reasons for deduping this logic relative to `getTelemetryProperties`, and I'll do a follow-up PR so the behavior change is very clear.  See this test case: https://github.com/microsoft/FluidFramework/pull/5308/files#diff-35c7882a903c065a87d17d968d0a90201eb945ab314d30a2608e6dba81c0f620R95

It wouldn't be wrong to push on this in this PR, but I'll be able to think more clearly on it with a fresh start rather than dragging this on.

This code has been very sloppy/optimistic, and I'm trying to be very deliberate about shoring it up.